compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1955,7 +1956,10 @@ trait Applications extends Compatibility {
     }
 
     def narrowByTypes(alts: List[TermRef], argTypes: List[Type], resultType: Type): List[TermRef] =
-      alts.filterConserve(isApplicableMethodRef(_, argTypes, resultType, ArgMatch.CompatibleCAP))
+      alts.filterConserve { alt =>
+        isApplicableMethodRef(alt, argTypes, resultType, ArgMatch.CompatibleCAP)
+        || isApplicableType(alt, argTypes, resultType)
+      }
```

Comment:
```suggestion
      alts.filterConserve {
        case expr: TermRef if expr.widenSingleton.isInstanceOf[ExprType] => isApplicableType(expr, argTypes, resultType)
        case alt => isApplicableMethodRef(alt, argTypes, resultType, ArgMatch.CompatibleCAP)
      }
```
Since a `TermRef` with a widen singleton of type `ExprType` will fail at `isApplicableMethodRef`, it might be possible to "by-pass" it, and not go into `isApplicableType` if we have a `MethodType` because it will just re-compute the result of `isApplicableMethodRef`